### PR TITLE
libgpg error: add patch to build v1.50 on Darwin

### PR DESCRIPTION
**Pull request by alalazo**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libassuan/package.py [hunk 1]
```diff
@@ -27,8 +27,17 @@ class Libassuan(AutotoolsPackage):
...
-        return [
-            "--enable-static",
-            "--enable-shared",
-            f"--with-libgpg-error-prefix={self.spec['libgpg-error'].prefix}",
-        ]
+        config_args = []
+
+        if self.spec.satisfies("@3 %apple-clang platform=darwin"):
+            # This is to avoid duplicate symbols errors on linking
+            config_args.append("CFLAGS=-fgnu89-inline")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/libassuan/package.py]: don't set CFLAGS :)

File: var/spack/repos/builtin/packages/libassuan/package.py [hunk 2]
```diff
@@ -27,8 +27,17 @@ class Libassuan(AutotoolsPackage):
...
 
     def configure_args(self):
-        return [
-            "--enable-static",
-            "--enable-shared",
-            f"--with-libgpg-error-prefix={self.spec['libgpg-error'].prefix}",
-        ]
+        config_args = []
+
+        if self.spec.satisfies("@3 %apple-clang platform=darwin"):
```
Comment (haampie) [inline var/spack/repos/builtin/packages/libassuan/package.py]: does this guard against use of apple-clang outside of darwin ;p

File: var/spack/repos/builtin/packages/libassuan/package.py [hunk 3]
```diff
@@ -26,6 +26,12 @@ class Libassuan(AutotoolsPackage):
 
     depends_on("libgpg-error@1.17:")
 
+    def flag_handler(self, name, flags):
+        if name == "cflags" and self.spec.satisfies("@3 platform=darwin"):
```
Comment (haampie) [inline var/spack/repos/builtin/packages/libassuan/package.py]: Is the platform a cause of the failure?

Comment (alalazo) [reply var/spack/repos/builtin/packages/libassuan/package.py]: Yes, or at least platform + Apple clang

**Review by haampie — APPROVED** [review]

**Pull request merged**